test/Dialect/Ibis/round-trip.mlir

```diff
@@ -1,56 +1,54 @@
 // RUN: circt-opt %s | circt-opt | FileCheck %s
 
-// CHECK-LABEL:  ibis.class @A {
+// CHECK-LABEL:  ibis.class @A(%this) {
 // CHECK-NEXT:    ibis.port.input @A_in : i1
 // CHECK-NEXT:    ibis.port.output @A_out : i1
 // CHECK-NEXT:  }
 
-// CHECK-LABEL:  ibis.class @C {
+// CHECK-LABEL:  ibis.class @C(%this) {
 // CHECK-NEXT:    ibis.port.input @C_in : i1
 // CHECK-NEXT:    ibis.port.output @C_out : i1
-// CHECK-NEXT:    ibis.instance @a, @A
-// CHECK-NEXT:    %true = hw.constant true
-// CHECK-NEXT:    ibis.port.write @C_in(%true) : i1
-// CHECK-NEXT:    %0 = ibis.port.read @C_out : i1
-// CHECK-NEXT:    ibis.instance.port.write @a::@A_in(%0) : i1
-// CHECK-NEXT:    %1 = ibis.instance.port.read @a::@A_out : i1
+// CHECK-NEXT:    %0 = ibis.instance @a, @A 
+// CHECK-NEXT:    %1 = ibis.get_parent %0 : !ibis.classref<@A>
+// CHECK-NEXT:    %2 = ibis.get_child @a : @A in %this : !ibis.classref<@C> 
 // CHECK-NEXT:    ibis.container @D {
-// CHECK-NEXT:      %true_0 = hw.constant true
-// CHECK-NEXT:      ibis.port.write @C_in(%true_0) : i1
-// CHECK-NEXT:      %2 = ibis.port.read @C_out : i1
-// CHECK-NEXT:      ibis.instance.port.write @a::@A_in(%2) : i1
-// CHECK-NEXT:      %3 = ibis.instance.port.read @a::@A_out : i1
+// CHECK-NEXT:      %3 = ibis.get_port %this, @C_in : !ibis.classref<@C> -> !ibis.portref<i1>
```

Comment:
Plz also add tests for getting the port of a parent and getting a port from a cousin.

Reply:
In this state, there isn't a method to get a port from a parent (which i also think we discussed during the last meeting, that we only need to access a sibling through the parent). This is an issue seeing as currently `get_port` requires a typed classref to extract a port.